administrator/components/com_joomlaupdate/models/default.php

```diff
@@ -289,109 +300,175 @@ public function purge()
+			if ($downloadInformation->exists)
 			{
-				// Passing false here -> download failed message
-				$response['basename'] = false;
+				$downloadInformation->done = true;
 
-				return $response;
+				return $downloadInformation;
 			}
+
+			// Save the download information to the session.
+			$session->set('com_joomlaupdate.download', $downloadInformation);
 		}
 
-		// Remove protocol, path and query string from URL
-		$basename = basename($packageURL);
+		// Get the information from the session and immediately clear it (guard against fatal errors).
+		$downloadInformation = $session->get('com_joomlaupdate.download', null);
+		$session->set('com_joomlaupdate.download', null);
 
-		if (strpos($basename, '?') !== false)
+		if ($downloadInformation === null)
 		{
-			$basename = substr($basename, 0, strpos($basename, '?'));
+			// I don't have any download information. I do not know what to do.
+			return null;
 		}
 
-		// Find the path to the temp directory and the local package.
-		$config   = JFactory::getConfig();
-		$tempdir  = (string) InputFilter::getInstance(array(), array(), 1, 1)->clean($config->get('tmp_path'), 'path');
-		$target   = $tempdir . '/' . $basename;
-		$response = array();
+		// Single part downloads get a very simple handling.
+		if ($forceSinglePart)
+		{
+			// Go through the mirrors until we find one that works or all have failed.
+			foreach ($downloadInformation->mirrors as $url)
+			{
+				$download = $this->downloadPackage($url, $downloadInformation->localFile);
 
-		// Do we have a cached file?
-		$exists = JFile::exists($target);
+				if ($download !== false)
+				{
+					break;
+				}
+			}
 
-		if (!$exists)
-		{
-			// Not there, let's fetch it.
-			$mirror = 0;
+			$downloadInformation->done  = $download !== false;
+			$downloadInformation->valid = $this->isChecksumValid($downloadInformation->localFile, $downloadInformation->updateInfo);
+
+			return $downloadInformation;
+		}
 
-			while (!($download = $this->downloadPackage($packageURL, $target)) && isset($sources[$mirror]))
+		// Chunked download, first part. Delete the local file.
+		if ($frag === 0)
+		{
+			if (File::exists($downloadInformation->localFile))
 			{
-				$name       = $sources[$mirror];
-				$packageURL = trim($name->url);
-				$mirror++;
+				@unlink($downloadInformation->localFile) || File::delete($downloadInformation->localFile);
 			}
 
-			$response['basename'] = $download;
+			JLog::add(JText::sprintf('COM_JOOMLAUPDATE_UPDATE_LOG_URL', $downloadInformation->url), JLog::INFO, 'Update');
 		}
-		else
+
+		// Calculate the download range
+		$from    = max($frag, 0) * self::CHUNK_LENGTH;
+		$to      = $from + self::CHUNK_LENGTH - 1;
+		$headers = array('Range' => sprintf('bytes=%u-%u', $from, $to));
+
+		try
 		{
-			// Is it a 0-byte file? If so, re-download please.
-			$filesize = @filesize($target);
+			JLog::add(JText::sprintf('COM_JOOMLAUPDATE_UPDATE_LOG_CHUNK', $frag + 1, $from, $to), JLog::INFO, 'Update');
```

Comment:
```suggestion
			JLog::add(JText::sprintf('COM_JOOMLAUPDATE_UPDATE_LOG_CHUNK', $frag, $from, $to), JLog::INFO, 'Update');
```
Possibly this is the reason for the wrong log mentioned in the comment https://github.com/joomla/joomla-cms/pull/38774#issuecomment-1249337469 ?